airflow/providers/google/cloud/hooks/spanner.py

```diff
@@ -70,6 +77,34 @@ def _get_client(self, project_id: str) -> Client:
             )
         return self._client
 
+    def _get_conn_params(self) -> tuple:
+        """Extract spanner database connection parameters."""
+        extras = self.get_connection(self.gcp_conn_id).extra_dejson
+        project_id = get_field(extras, "project_id") or self.project_id
+        instance_id = get_field(extras, "instance_id")
+        database_id = get_field(extras, "database_id")
```

Comment:
Are these fields always there? Is their absence possible and what happens in this case?

Reply:
If their absence then these variables are None and code below throws Exception.

Reply:
LGTM, thanks